### PR TITLE
Add config model and crd client

### PR DESCRIPTION
**Pull request by xiaolanz**
issue #15 
Add broker config model under pkg/model/config
Move obsi model under pkg/model/obs
Add crd client under pkg/platform/kube/crd

Unittests and integration tests.
The code was adopted from istio/pilot/

### CODE REVIEW

File: pkg/platform/kube/crd/client.go [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+
+// RegisterResources sends a request to create CRDs and waits for them to initialize
+func (cl *Client) RegisterResources() error {
+	clientset, err := apiextensionsclient.NewForConfig(cl.restconfig)
+	if err != nil {
+		return err
+	}
+
+	for _, schema := range cl.descriptor {
+		name := schema.Plural + "." + config.IstioAPIGroup
```
Comment (kyessenov) [inline pkg/platform/kube/crd/client.go]: you should remove "-" from all kubernetes names. it confuses `kubectl` for some combination of apiserver/kubectl.

**Comment by kyessenov** [comment]
Looks fine (lots of familiar code :)) Maybe put a backreference to pilot so that the origin does not get lost, and we get some bug fixes flowing both ways.

File: pkg/platform/kube/crd/client.go (hunk 1 above)
Comment (ayj) [reply pkg/platform/kube/crd/client.go]: https://github.com/istio/pilot/issues/1107

File: pkg/platform/kube/crd/client.go [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+		rd := &apiextensionsv1beta1.CustomResourceDefinition{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name: name,
+			},
+			Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
+				Group:   config.IstioAPIGroup,
+				Version: config.IstioAPIVersion,
+				Scope:   apiextensionsv1beta1.NamespaceScoped,
+				Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
+					Singular: schema.Type,
```
Comment (xiaolanz) [inline pkg/platform/kube/crd/client.go]: @ayj @kyessenov I set the Singular name specifically here (without it I ran into other naming problems), so I think '-' should just work.

**Comment by xiaolanz** [comment]
PTAL
disable linter and code coverage in prow due to none trivial fixes. Will fix in another  PR.

**Pull request merged**